### PR TITLE
Add l10n_it_sepa_bonifici

### PR DESCRIPTION
**Pull request by alessandrocamilli**

### CODE REVIEW

**Comment by alessandrocamilli** [comment]
Remove work in progress please :-)

File: l10n_it_sepa_bonifici/i18n/it.po [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+"Language-Team: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: \n"
+"Plural-Forms: \n"
+
+#: field:banking.export.sepa.cbi.wizard,batch_booking:0
+#. module: l10n_it_sepa_bonifici
+msgid "Batch Booking"
+msgstr "Batch Booking"
```
Comment (OpenCode) [inline l10n_it_sepa_bonifici/i18n/it.po]: Is this untranslatable?

File: l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+        return True
+
+    @api.model
+    def generate_creditor_scheme_identification(
+            self, parent_node, identification, identification_label,
+            eval_ctx, scheme_name_proprietary, gen_args):
+        #
+        # CBI logic modified for not try to add other info
+        #
+        '''
```
Comment (OpenCode) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py]: Can you remove commented lines?

File: l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+                self.generate_start_payment_info_block(
+                    pain_root,
+                    "self.payment_order_ids[0].reference + '-' "
+                    "+ requested_date.replace('-', '')  + '-' + priority",
+                    priority, False, False, requested_date, {
+                        'self': self,
+                        'priority': priority,
+                        'requested_date': requested_date,
+                    }, gen_args)
+            # >>>>>>>>>
```
Comment (OpenCode) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py]: Remove commented line

File: l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+        csi_scheme_name_proprietary = etree.SubElement(
+            csi_scheme_name, 'Prtry')
+        csi_scheme_name_proprietary.text = scheme_name_proprietary
+        '''
+        return True
+
+    @api.multi
+    def create_sepa(self):
+        """Creates the SEPA Credit Transfer file. That's the important code!"""
+        sepa_export = self[0]
```
Comment (OpenCode) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py]: Use self.ensure_one() if you want to be sure to have only one recordset

File: l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+                control_sum_2_5.text = '%.2f' % amount_control_sum_2_5
+        if pain_flavor in pain_03_to_05:
+            nb_of_transactions_1_6.text = str(transactions_count_1_6)
+            control_sum_1_7.text = '%.2f' % amount_control_sum_1_7
+        else:
+            nb_of_transactions_1_6.text = str(transactions_count_1_6)
+            control_sum_1_7.text = '%.2f' % amount_control_sum_1_7
+
+        # CBI required
+        # >> v8
```
Comment (OpenCode) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py]: Remove commented lines

File: l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py [hunk 6]
```diff
@@ -0,0 +1,380 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning
```
Comment (OpenCode) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py]: Replace generic Warning with UserError

File: l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py [hunk 7]
```diff
@@ -0,0 +1,380 @@
...
+        return True
+
+    @api.model
+    def generate_creditor_scheme_identification(
+            self, parent_node, identification, identification_label,
+            eval_ctx, scheme_name_proprietary, gen_args):
+        #
+        # CBI logic modified for not try to add other info
+        #
+        '''
```
Comment (OpenCode) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py]: Remove commented lines

File: l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py [hunk 8]
```diff
@@ -0,0 +1,380 @@
...
+        csi_scheme_name_proprietary = etree.SubElement(
+            csi_scheme_name, 'Prtry')
+        csi_scheme_name_proprietary.text = scheme_name_proprietary
+        '''
+        return True
+
+    @api.multi
+    def create_sepa(self):
+        """Creates the SEPA Credit Transfer file. That's the important code!"""
+        sepa_export = self[0]
```
Comment (OpenCode) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py]: Use self.ensure_one() if you want to be sure to have only one recordset

**Comment by eLBati** [comment]
Thanks @alessandrocamilli 
Please see failing travis checks
```
======== Testing test_flake8 ========
l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py:69:69: E126 continuation line over-indented for hanging indent
l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py:86:21: E126 continuation line over-indented for hanging indent
l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py:266:72: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py:278:60: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py:279:35: E131 continuation line unaligned for hanging indent
l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py:281:80: E501 line too long (84 > 79 characters)
l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py:281:85: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py:287:56: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:29:1: F401 'base64' imported but unused
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:81:13: E128 continuation line under-indented for visual indent
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:82:1: W293 blank line contains whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:87:69: E126 continuation line over-indented for hanging indent
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:95:59: E231 missing whitespace after ','
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:101:49: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:107:18: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:115:21: E126 continuation line over-indented for hanging indent
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:120:74: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:123:79: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:129:51: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:132:1: W293 blank line contains whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:181:66: E502 the backslash is redundant between brackets
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:182:17: E131 continuation line unaligned for hanging indent
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:189:80: E501 line too long (82 > 79 characters)
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:192:80: E501 line too long (85 > 79 characters)
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:199:52: E261 at least two spaces before inline comment
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:199:53: E262 inline comment should start with '# '
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:201:1: W293 blank line contains whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:236:70: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:238:62: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:242:66: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:243:58: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:274:70: E261 at least two spaces before inline comment
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:274:80: E501 line too long (80 > 79 characters)
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:299:60: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:300:35: E131 continuation line unaligned for hanging indent
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:309:72: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:321:60: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:322:35: E131 continuation line unaligned for hanging indent
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:324:80: E501 line too long (84 > 79 characters)
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:324:85: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:330:56: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:352:9: E265 block comment should start with '# '
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:355:73: W291 trailing whitespace
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:362:1: W293 blank line contains whitespace
======== Testing test_pylint ========
************* Module l10n_it_sepa_bonifici
l10n_it_sepa_bonifici/__init__.py:5: [W0403(relative-import), ] Relative import 'wizard', should be 'l10n_it_sepa_bonifici.wizard'
************* Module l10n_it_sepa_bonifici.wizard
l10n_it_sepa_bonifici/wizard/__init__.py:5: [W0403(relative-import), ] Relative import 'export_sepa_cbi', should be 'l10n_it_sepa_bonifici.wizard.export_sepa_cbi'
l10n_it_sepa_bonifici/wizard/__init__.py:6: [W0403(relative-import), ] Relative import 'export_sepa_cbi_estero', should be 'l10n_it_sepa_bonifici.wizard.export_sepa_cbi_estero'
************* Module l10n_it_sepa_bonifici.wizard.export_sepa_cbi_estero
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:358: [E1601(print-statement), BankingExportSepaCbiEsteroWizard.create_sepa] print statement used
From https://github.com/OCA/l10n-italy
 * [new branch]      8.0        -> origin/8.0
Start lint check just in modules changed
************* Module l10n_it_sepa_bonifici.wizard.export_sepa_cbi
l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py:7: [R8101(openerp-exception-warning), ] Import `Warning` should be renamed as UserError `from openerp.exceptions import Warning as UserError`
************* Module l10n_it_sepa_bonifici.wizard.export_sepa_cbi_estero
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:1: [C8201(no-utf8-coding-comment), ] No UTF-8 coding comment found: Use `# coding: utf-8` or `# -*- coding: utf-8 -*-`
l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py:25: [R8101(openerp-exception-warning), ] Import `Warning` should be renamed as UserError `from openerp.exceptions import Warning as UserError`
pylint expected errors 0, found 4!
```

File: l10n_it_sepa_bonifici/README.rst [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+bonifico in Italia, come ad esempio la presenza obbligatoria dell’ABI della
+banca di addebito contenuto nel campo “MmbId” che è invece facoltativo
+nel tracciato ISO. 
+
+Configuration
+=============
+
+Configure the CUC code and the Issuer code (must be "CBI").
+
+.. image:: /l10n_it_sepa_bonifici/static/company_setting.png
```
Comment (eLBati) [inline l10n_it_sepa_bonifici/README.rst]: Please use ```.. figure::``` to also support apps.odoo.com
See https://raw.githubusercontent.com/OCA/account-financial-reporting/9.0/mis_builder/README.rst

**Comment by alessandrocamilli** [comment]
@eLBati after last push, where can i find the updated list generated by travis?

**Comment by archetipo** [comment]
@alessandrocamilli 
[here](https://github.com/OCA/l10n-italy/pull/218#partial-pull-merging) on line  **continuous-integration/travis-ci/pr** click on Details , on new page you can see a list of travis checks, in this case first item is red , click on and see the travis log.

**Comment by alessandrocamilli** [comment]
@archetipo thanks!

**Comment by alessandrocamilli** [comment]
Travis is now ok. 
Can you see to approve it for merge?
Thanks :-)

**Comment by tafaRU** [comment]
@alessandrocamilli, thanks a lot for your great work!
Could you also please address @OpenCode remarks?

Thank you in advance.

File: l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py (hunk 2 above)
Comment (alessandrocamilli) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py]: See commit https://github.com/alessandrocamilli/l10n-italy/commit/a8524ae80adbfa6ebb5e3af3f3a8f0bc880ee4c7

File: l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py (hunk 8 above)
Comment (alessandrocamilli) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py]: See commit https://github.com/alessandrocamilli/l10n-italy/commit/a8524ae80adbfa6ebb5e3af3f3a8f0bc880ee4c7

File: l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py (hunk 7 above)
Comment (alessandrocamilli) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py]: Done https://github.com/alessandrocamilli/l10n-italy/commit/a8524ae80adbfa6ebb5e3af3f3a8f0bc880ee4c7

File: l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py (hunk 6 above)
Comment (alessandrocamilli) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py]: done https://github.com/OCA/l10n-italy/pull/218/commits/2ea0c257a84cd7e6d4231c16b44ccb768b74fed9

File: l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py (hunk 5 above)
Comment (alessandrocamilli) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py]: done https://github.com/OCA/l10n-italy/pull/218/commits/a8524ae80adbfa6ebb5e3af3f3a8f0bc880ee4c7

File: l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py (hunk 4 above)
Comment (alessandrocamilli) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py]: done https://github.com/OCA/l10n-italy/pull/218/commits/0f87c69e45293cd81a88b47577dd13bb764ef2dc

File: l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py (hunk 3 above)
Comment (alessandrocamilli) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py]: Done https://github.com/OCA/l10n-italy/pull/218/commits/a8524ae80adbfa6ebb5e3af3f3a8f0bc880ee4c7

File: l10n_it_sepa_bonifici/__init__.py [hunk 10]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# © 2016 Alessandro Camilli <alessandro.camilli@openforce.it>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from . import wizard
```
Comment (OpenCode) [inline l10n_it_sepa_bonifici/__init__.py]: PEP8 W391 blank line at end of file

File: l10n_it_sepa_bonifici/__openerp__.py [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+    'depends': [
+        'account_banking_pain_base',
+    ],
+    'data': [
+        'wizard/export_sepa_cbi_view.xml',
+        'wizard/export_sepa_cbi_estero_view.xml',
+        'data/payment_type_sepa_cbi.xml',
+        ],
+    'installable': True
+}
```
Comment (OpenCode) [inline l10n_it_sepa_bonifici/__openerp__.py]: PEP8 W391 blank line at end of file

File: l10n_it_sepa_bonifici/wizard/__init__.py [hunk 12]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
+# © 2016 Alessandro Camilli <alessandro.camilli@openforce.it>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from . import export_sepa_cbi
+from . import export_sepa_cbi_estero
```
Comment (OpenCode) [inline l10n_it_sepa_bonifici/wizard/__init__.py]: PEP8 W391 blank line at end of file

File: l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py [hunk 13]
```diff
@@ -0,0 +1,307 @@
...
+        for order in self.payment_order_ids:
+            workflow.trg_validate(
+                self._uid, 'payment.order', order.id, 'done', self._cr)
+            self.env['ir.attachment'].create({
+                'res_model': 'payment.order',
+                'res_id': order.id,
+                'name': self.filename,
+                'datas': self.file,
+                })
+        return True
```
Comment (OpenCode) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi.py]: PEP8 W391 blank line at end of file

File: l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py [hunk 14]
```diff
@@ -0,0 +1,383 @@
...
+        # Remove the duplicate node  LclInstrm in payment
+        # CtrlSum_node = xml_root.xpath('//PmtInf//LclInstrm')[0] >CBI required
+        # CtrlSum_node.getparent().remove(CtrlSum_node) You can
+        # remove node only from parent
+        # Remove the duplicate node  CtrlSum in payment
+        # CtrlSum_node = xml_root.xpath('//PmtInf//SeqTp')[0] >> CBI required
+        # CtrlSum_node.getparent().remove(CtrlSum_node) >> You can remove
+        # node only from parent
+        # >>>>>>>>>
+        # print(etree.tostring(xml_root, pretty_print=True))
```
Comment (OpenCode) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py]: Is this useful?

Comment (alessandrocamilli) [inline l10n_it_sepa_bonifici/wizard/export_sepa_cbi_estero.py]: I want to left the print for the future debug.
I removed the other comment lines

**Comment by eLBati** [comment]
@alessandrocamilli vedi https://github.com/alessandrocamilli/l10n-italy/pull/9

File: l10n_it_sepa_bonifici/i18n/it.po [hunk 15]
```diff
@@ -0,0 +1,165 @@
+# Translation of Odoo Server.
```
Comment (OpenCode) [inline l10n_it_sepa_bonifici/i18n/it.po]: A lot of strings are not translated

Comment (eLBati) [inline l10n_it_sepa_bonifici/i18n/it.po]: @OpenCode farei il merge così com'è e poi tradurrei tramite transifex. ok per te?

Comment (OpenCode) [reply l10n_it_sepa_bonifici/i18n/it.po]: Ok

**Comment by eLBati** [comment]
closing in favour of https://github.com/OCA/l10n-italy/pull/389

**Pull request closed without merging**